polars/polars-time/src/windows/groupby.rs

```diff
@@ -461,38 +461,20 @@ pub(crate) fn groupby_values_iter<'a>(
     tu: TimeUnit,
     tz: Option<Tz>,
 ) -> Box<dyn TrustedLen<Item = PolarsResult<(IdxSize, IdxSize)>> + 'a> {
-    let mut offset = period;
-    offset.negative = !period.negative;
-    if offset.duration_ns() > 0 {
+    if period.duration_ns() > 0 {
         // t is at the right endpoint of the window
+        let mut offset = period;
+        offset.negative = true;
         let iter =
             groupby_values_iter_full_lookbehind(period, offset, time, closed_window, tu, tz, 0);
         Box::new(iter)
-    } else if closed_window == ClosedWindow::Right || closed_window == ClosedWindow::None {
-        // only lookahead
-        let iter = groupby_values_iter_full_lookahead(
-            period,
-            offset,
-            time,
-            closed_window,
-            tu,
-            tz,
-            0,
-            None,
-        );
+    } else if matches!(closed_window, ClosedWindow::Both) {
+        // window only contains t
+        let iter = time.iter().enumerate().map(|(i, _)| Ok((i as IdxSize, 1)));
```

Comment:
We don't have to iterator `time` in that case. It is cheaper to iterate `0..time.len()`